aws_lambda_builders/workflows/ruby_bundler/bundler.py

```diff
@@ -51,7 +51,16 @@ def run(self, args, cwd=None):
         out, _ = p.communicate()
 
         if p.returncode != 0:
-            # Bundler has relevant information in stdout, not stderr.
-            raise BundlerExecutionError(message=out.decode("utf8").strip())
+            if p.returncode == 10:
```

Comment:
Is this documented?

Reply:
Although not explicitly documented by Bundler, the Bundler error codes can be seen [here](https://github.com/rubygems/bundler/blob/3f0638c6c8d340c2f2405ecb84eb3b39c433e36e/lib/bundler/errors.rb#L36).

Reply:
I see; please add a comment for reference to others as this is non-standard.

Tangentially, could this be handled in a more standard way (similar to https://github.com/aws/aws-lambda-builders/pull/243)? It seems Bundler always expects a `Gemfile` and isn't meaningful without it.

Reply:
We could maybe put the `10` in a constant like `GEMFILE_NOT_FOUND` (this would also remove the need for the comment at line 55).

You could put your link to error codes on the line before the constant.

Reply:
A discussion was had with the team regarding this change. This solution is non-standard because of the way Bundler behaves. Doing a similar approach to Python pip is dangerous because of Bundler's ability to search for a Gemfile in parent directories, meaning that it could break for users with non-standard project structures. Although unconventional, this approach was deemed safer.

I will go ahead and make the changes that Mathieu suggested to improve the clarity.

Reply:
Makes sense, thanks for the explanation. 👍